test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2439,9 +2439,41 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
         }
     }
 
+    enum {
+        SEND_PREREG_MEMH = UCS_BIT(0),
+        RECV_PREREG_MEMH = UCS_BIT(1)
+    };
+
+    void save_and_reset_mds(entity &e, std::vector<uct_md_h> *mds)
+    {
+        if (mds == NULL) {
+            return;
+        }
+
+        ucp_tl_md_t *tl_md;
+        ucs_carray_for_each(tl_md, e.ucph()->tl_mds, e.ucph()->num_mds) {
+            mds->push_back(tl_md->md);
+            tl_md->md = NULL;
+        }
+    }
+
+    void restore_mds(entity &e, std::vector<uct_md_h> *mds)
+    {
+        if (mds == NULL) {
+            return;
+        }
+
+        unsigned i = 0;
+        ucp_tl_md_t *tl_md;
+        ucs_carray_for_each(tl_md, e.ucph()->tl_mds, e.ucph()->num_mds) {
+            tl_md->md = (*mds)[i++];
+        }
+    }
+
     void test_am_send_recv(size_t size, size_t hdr_size = 0ul,
                            size_t num_iters = m_num_iters,
-                           bool recv_prereg_memh = false,
+                           uint8_t prereg_memh_flags = 0,
```

Comment:
bool send_prereg_memh
bool recv_prereg_memh
assert not both are true + comment why